Design/WorkspacePropertyWithIndex.md

```diff
@@ -0,0 +1,95 @@
+# WorkspacePropertyWithIndex Type
+
+This document relates to an algorithm property type which can be used to specify a workspace and desired input index type(s). Index translation will will be handled by the new `IndexInfo` (design [here](https://github.com/mantidproject/documents/blob/master/Design/spectrum_number_and_workspace_index_abstraction.md)) object internally. Badly unit tested code and inconsistencies in validating user indices would be absorbed by this type. This design complements `IndexInfo` and reinforces some of the design concepts. **Note**:The property would then return the workspace and workspace indices (after automatic translation).
```

Comment:
Consider adding another note: This is about index properties that *are used to obtain a workspace index*. This property does *not* cater for applications that directly need, e.g., detector IDs.